sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -188,6 +189,87 @@ case class Rint(child: Expression) extends UnaryMathExpression(math.rint, "ROUND
   override def funcName: String = "rint"
 }
 
+case class ShiftRightUnsigned(left: Expression, right: Expression) extends Expression{
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+    if (!(left.dataType.isInstanceOf[ByteType]
```

Comment:
almost the same thing for the performance point the view? any way it might be more concise by using seq.